### PR TITLE
valve does not inherit node

### PR DESCRIPTION
**Pull request by 3nids**
doing a PR so we centralize the discussion

### CODE REVIEW

File: update/delta/delta_0.1.1.sql [hunk 1]
```diff
@@ -0,0 +1,68 @@
+ALTER TABLE qwat_od.valve ADD COLUMN fk_district integer;
```
Comment (3nids) [inline update/delta/delta_0.1.1.sql]: oh. It seems this file is no more necessary

File: update/delta/delta_1.1.0.sql [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+            PERFORM qwat_od.fn_valve_set_orientation(r.id);
+        END LOOP;
+
+        RETURN NULL;
+    END;
+$BODY$
+LANGUAGE plpgsql;
+COMMENT ON FUNCTION qwat_od.ft_valve_pipe_update() IS 'Trigger: when moving or deleting a pipe, reassign the pipe to all valves connected to the old pipe and recalculate valve orientation. Do an AFTER trigger since it will update valve after updating the node.';
+/* WHEN THE PIPE MOVES */
+CREATE TRIGGER tr_valve_pipe_update
```
Comment (3nids) [inline update/delta/delta_1.1.0.sql]: ERROR:  trigger "tr_valve_pipe_update" for relation "pipe" already exists

**Comment by haubourg** [comment]
@sylvainbeo @3nids Hi guys, anything new on this side? 
We need to achieve that part to work on a cleaner and functional qgs project . 
Thanks for your lights.

**Comment by 3nids** [comment]
There is still one comment pending.
Otherwise, waiting on the merge!

**Comment by sylvainbeo** [comment]
Yes, i will have a look on wednesday.


2016-10-31 16:42 GMT+01:00 Denis Rouzaud <notifications@github.com>:

**Comment by sylvainbeo** [comment]
We merge ?

**Comment by 3nids** [comment]
tested on demo data?

**Comment by sylvainbeo** [comment]
Ah. I thought you already did it.

**Comment by 3nids** [comment]
That would be the second point on the todo list in https://github.com/qwat/QWAT/issues/98
I think we should achieve this before merging this one.

**Comment by 3nids** [comment]
are you waiting on an update of demo data?
if it is the case, I can test this on my DB here and we can merge afterwards.

Would still be nice to have update of version in the DB in the delta.

**Comment by sylvainbeo** [comment]
Tested on the demo data. Seems ok.

**Comment by sylvainbeo** [comment]
Denis, can you have a test on your data to confirm  ?

- launch the delta
- rebuild the views

**Comment by 3nids** [comment]
I'm on it

**Comment by 3nids** [comment]
delta script went fine in 10:35 min (!)
otherwise, I couldn't recreate the views, I get the error `ERROR:  column "fk_district" specified more than once`.
Not sure where does it come from at the moment. Looking....

File: ordinary_data/views/export/search_view.sql [hunk 3]
```diff
@@ -31,10 +31,4 @@ CREATE OR REPLACE VIEW qwat_od.vw_search_view AS
 		'Compteur' as layer_name,
 		COALESCE(district_prefix||'_')||meter.identification AS search_text,
 		ST_Force2d(meter.geometry) AS geometry
-		FROM qwat_od.vw_export_meter meter
-
-	UNION SELECT
-		'Vannes' as layer_name,
-		valve_function_value_fr || ' ' || identification || ' ' || district_name as search_text,
-		ST_Force2d(geometry) AS geometry
-		FROM qwat_od.vw_export_valve WHERE identification IS NOT NULL;
```
Comment (sylvainbeo) [inline ordinary_data/views/export/search_view.sql]: Ok, i restore it back.

**Comment by 3nids** [comment]
OK, everything went fine.
I just had to rename views in QGIS project.
Will be going for 1-2 days testing here before going on final production, but +1 to merge!
yeah!!!

**Comment by sylvainbeo** [comment]
Excellent !
For the QGIS project, did you do it inside qgis, or by editing the project file ? I wasn't so hard for me, just a bit fastidious.

**Comment by 3nids** [comment]
Also, in the delta file, add the update of the hardcoded version number.

**Comment by 3nids** [comment]
Yeah just rename the 2 valves views in the missing layer dialog when opening the project. Quite easy, but you have to know what has been changed!

**Comment by 3nids** [comment]
I just added missing fields for labels.
I let you rebase as I am not sure what you expect in the trigger definition.

**Comment by sylvainbeo** [comment]
You can do the rebase if you want. Triggers simplifications are well compartimented, and i will take in account those changes in the lvl_triggers branch.

**Comment by 3nids** [comment]
OK, can't do it now, but later today.

Still missing update of system version, let me know if you add it, otherwise will do this afternoon.

Aslo, can you let me know how to properly tag the commit (never done it before).

**Comment by sylvainbeo** [comment]
I always did it via github, following this : https://help.github.com/articles/creating-releases/
Easy to do. Just follow the steps. 
You can do it directly via git: https://git-scm.com/book/en/v2/Git-Basics-Tagging

**Comment by 3nids** [comment]
valve miss handling of alternative geometries...fixing this

**Comment by 3nids** [comment]
and it seems calculation of orientation is missing too! @sylvainbeo I thought you implemented it, did you?

**Comment by 3nids** [comment]
@sylvainbeo also, I have troubles at rebasing, may I let you do it?

**Comment by 3nids** [comment]
@sylvainbeo rebase successful....was quite painful, now lights are green :satisfied:

Some fixes to conformity tests (see https://github.com/qwat/QWAT/issues/98), but I would wait on fixing the remaining ones before merging this PR.

This version 1.1.0 is in production at the SIGE since this morning without any issues raised up to now...
We are on the good path :sweat_smile:

**Comment by sylvainbeo** [comment]
Excellent job !

**Comment by 3nids** [comment]
hey.....did you just push force???
I had just completed a rebase.....

**Comment by sylvainbeo** [comment]
No. I just did a simple push to update the delta (with the 3 functions deletion)

File: update/delta/delta_1.1.0.sql [hunk 4]
```diff
@@ -309,8 +309,8 @@ DROP TRIGGER valve_node_set_type ON qwat_od.valve;
 DROP FUNCTION qwat_od.ft_valve_node_set_type();
 
 -- We need to manually remove those 3 triggers functions, as they cannot be removed automatically by the rewrite views process
-DROP FUNCTION qwat_od.ft_element_valve_delete();
-DROP FUNCTION qwat_od.ft_element_valve_insert();
-DROP FUNCTION qwat_od.ft_element_valve_update();
+DROP FUNCTION qwat_od.ft_element_valve_delete() cascade;
+DROP FUNCTION qwat_od.ft_element_valve_insert() cascade;
+DROP FUNCTION qwat_od.ft_element_valve_update() cascade;
```
Comment (3nids) [inline update/delta/delta_1.1.0.sql]: oh no let's don't do cascades...if there is anything associated outside of qwat, it is problematic.
better write down all need drops.

Comment (sylvainbeo) [inline update/delta/delta_1.1.0.sql]: Ok. i do it.

**Comment by 3nids** [comment]
I just pushed force again with the rebase (without the cascades)

**Comment by 3nids** [comment]
sorry for the troubles, I hope you're doing fine with cherry-pic!

**Comment by 3nids** [comment]
???? no push force? how does it happen that you can overwrite my rebase?????

**Comment by 3nids** [comment]
I rebased and pushed force again.....please fetch and cherry-pick any extra commit on top of origin/valve_exit_node....

**Comment by sylvainbeo** [comment]
Well i did not push force.. Anyway, all is ok now, you can merge in master.

**Comment by 3nids** [comment]
thanks a lot, glad to see this merged!

**Comment by sylvainbeo** [comment]
So do i ! Great work to you as well.

**Pull request merged**